teleport.yaml

```diff
@@ -35,32 +37,62 @@ pipeline:
   - uses: git-checkout
     with:
       repository: https://github.com/gravitational/teleport
-      expected-commit: dc5837102a82fdfc4807a8c760839f4ca4be08a9
+      expected-commit: a5c84e4b74f1da43a44bd6c18ae184f612fb26f6
       tag: v${{package.version}}
 
   - uses: go/bump
     with:
       deps: github.com/golang-jwt/jwt/v4@v4.5.1
 
   - runs: |
-      # https://github.com/gravitational/teleport#building-teleport
       mkdir -p "${{targets.contextdir}}"/var/lib/teleport
       mkdir -p "${{targets.contextdir}}"/usr/local/bin
 
-      # This build requires the stable version of rust, managed by rustup, because it requires a few other toolchains too.
       rustup install stable
+      rustup default stable
+      ARCH=$(uname -m)
+      export PATH="$HOME/.rustup/toolchains/stable-${ARCH}-unknown-linux-gnu/bin:$PATH"
       rustup target add wasm32-unknown-unknown
 
-      # This is a bit of a hack, but it's the easiest way to get the right version of rustc and cargo in the path.
-      export PATH="$HOME/.rustup/toolchains/stable-${{host.triplet.rust}}/bin:$PATH"
-      make full
+      pnpm config set package-import-method copy
 
-      # create me a for loop that iterates over the binaries in build/
-      # and installs them to "${{targets.contextdir}}"/usr/local/bin
-      for bin in build/*; do
-        echo "Installing $bin to /usr/local/bin"
-        install -Dm755 $bin -t "${{targets.contextdir}}"/usr/local/bin
-      done
+      # Install dependencies and build web assets
+      make ensure-js-deps
+      make ensure-webassets
+
+  - uses: go/build
+    with:
+      packages: ./tool/teleport
+      output: teleport
+      tags: webassets_embed,kustomize_disable_go_plugin_support
+
+  - uses: go/build
+    with:
+      packages: ./tool/tctl
+      output: tctl
+      tags: kustomize_disable_go_plugin_support
+
+  - uses: go/build
+    with:
+      packages: ./tool/tsh
+      output: tsh
+      tags: kustomize_disable_go_plugin_support
+
+  - uses: go/build
+    with:
+      packages: ./tool/tbot
+      output: tbot
+      tags: kustomize_disable_go_plugin_support
+
+  - uses: go/build
+    with:
+      packages: ./tool/teleport-update
+      output: teleport-update
+      tags: kustomize_disable_go_plugin_support
+
+  - runs: |
+      cd tool/fdpass-teleport && cargo build --release --locked
```

Comment:
```suggestion
      cd tool/fdpass-teleport && cargo auditable build --release --locked
```

Reply:
Note that one can check it was built correctly with:

```
$ rust-audit-info /usr/local/bin/fdpass-teleport 
No audit data found in the binary! Was it built with 'cargo auditable'?
```

Also I wonder if cargo auditable should ship wrapper in /usr/local/bin like with do with openssf compiler options; such that auditing is picked up automatically. Will follow up on this separately.